app/src/main/java/com/fantastic/bookxchange/fragments/AddBookSearchingFragment.java

```diff
@@ -0,0 +1,197 @@
+package com.fantastic.bookxchange.fragments;
+
+import android.content.Context;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.SearchView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+import android.widget.Toast;
+
+import com.fantastic.bookxchange.R;
+import com.fantastic.bookxchange.adapters.BookTestAdapter;
+import com.fantastic.bookxchange.api.BookClient;
+import com.fantastic.bookxchange.api.JsonKeys;
+import com.fantastic.bookxchange.models.BookTest;
+import com.loopj.android.http.JsonHttpResponseHandler;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+import cz.msebera.android.httpclient.Header;
+import retrofit2.http.PUT;
+
+
+public class AddBookSearchingFragment extends Fragment {
+
+    private ListView lvBooks;
+    private BookTestAdapter bookTestAdapter;
+    private BookClient client;
+    private ArrayList<BookTest> mBooks;
+    private ProgressBar mProgressBar;
+
+
+    private OnFragmentInteractionListener mListener;
+
+    public AddBookSearchingFragment() {
+        // Required empty public constructor
+    }
+
+
+    public static AddBookSearchingFragment newInstance() {
+        AddBookSearchingFragment fragment = new AddBookSearchingFragment();
+        Bundle args = new Bundle();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setHasOptionsMenu(true);
+        if (getArguments() != null) {
+
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+
+        View view = inflater.inflate(R.layout.fragment_add_book_searching, container, false);
+
+        lvBooks = view.findViewById(R.id.lvBooks);
+        mBooks = new ArrayList<>();
+        bookTestAdapter = new BookTestAdapter(getContext(), mBooks);
+
+        setupFooter();
+
+        lvBooks.setOnItemClickListener(new AdapterView.OnItemClickListener() {
```

Comment:
Probably is better to have some kind of button on the right side of the row. That way the user can also see the detail view of the book clicking on the row.

Reply:
ok, great, I'll do it too

Reply:
Thanks for your feedback